File: csharp/933.number-of-recent-calls.cs
/*
 * @lc app=leetcode id=933 lang=csharp
 *
 * [933] Number of Recent Calls
 */

using System.Collections.Generic;

// @lc code=start
public class RecentCounter {

    private Queue<int> Queue;

    public RecentCounter() {
        Queue = new Queue<int>();
    }
    
    public int Ping(int t) {
        Queue.Enqueue(t);
        while (Queue.Peek() < t - 3000) {
            Queue.Dequeue();
        }
        return Queue.Count;
    }
}

/**
 * Your RecentCounter object will be instantiated and called as such:
 * RecentCounter obj = new RecentCounter();
 * int param_1 = obj.Ping(t);
 */
// @lc code=end

File: java/933.number-of-recent-calls.java
/*
 * @lc app=leetcode id=933 lang=java
 *
 * [933] Number of Recent Calls
 */

import java.util.ArrayDeque;
import java.util.Queue;

// @lc code=start
class RecentCounter {

    private final Queue<Integer> queue;

    public RecentCounter() {
        queue = new ArrayDeque<>();
    }
    
    public int ping(int t) {
        while (queue.size() > 0 && queue.peek() < t - 3000) {
            queue.poll();
        }
        queue.offer(t);
        return queue.size();
    }
}

/**
 * Your RecentCounter object will be instantiated and called as such:
 * RecentCounter obj = new RecentCounter();
 * int param_1 = obj.ping(t);
 */
// @lc code=end

